### PR TITLE
Fixes #20999 - Reset env/cv for Activation Keys

### PR DESCRIPTION
**Pull request by parthaa**
This commit allows one unset the environment and content view for
activation keys via the UI.

### CODE REVIEW

**Comment by parthaa** [comment]
Behavior
- Added the reset button 'X' next to the content view selection box to reset the CV for the activation key
- If you unselect an environment, it will reset the content view also. 

testak

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details-info.controller.js [hunk 1]
```diff
@@ -61,6 +67,12 @@ angular.module('Bastion.activation-keys').controller('ActivationKeyDetailsInfoCo
             $scope.disableEnvironmentSelection = false;
         };
 
+        $scope.clearContentView = function (activationKey) {
+            activationKey["content_view_id"] = null;
+            activationKey["content_view"] = null;
+            $scope.saveContentView(activationKey);
+        };
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details-info.controller.js]: shoudln't this unset the lifecycle environment too?

Comment (parthaa) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details-info.controller.js]: @jlsherrill good point. Will look into it. One of the issues is that with the current environment selector resetting is harder than setting `activationKey.environment = null`. 
@jlsherrill @rohoover should I just get rid of it ?

**Comment by parthaa** [comment]
@jlsherrill @jturel @rohoover after giving a valiant fight to vanquish the evil env selector I decided its easier to  just reset the existing selector. So here is the new behavior. (Too many challenges with the tree design especially on the AK edit page.)

ak-new

**Comment by parthaa** [comment]
In short reseting Content View will wipe out environment  and Unchecking env will reset the content view

**Comment by jturel** [comment]
When content view has been cleared clicking into the dropdown will throw an error (Cannot read property id of null)

https://github.com/parthaa/katello/blob/d580e1f0517037cffa83815b24d5617722a74bb5/engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details-info.controller.js#L118

**Comment by parthaa** [comment]
> When content view has been cleared clicking into the dropdown will throw an error (Cannot read property id of null)
Ok updated the behavior a little. Basically you cant select the content view unless you have an environment selected. Check the new version

aka-3

**Comment by parthaa** [comment]
[test]

**Comment by parthaa** [comment]
[test katello]

**Pull request merged**